e2e/cypress/integration/messaging/pinned_posts_spec.js

```diff
@@ -48,4 +49,146 @@ describe('Messaging', () => {
+        });
+    });
+
+    it('MM-T2169 Un-pinning a post in center also removes badge from *search results* RHS', () => {
+        // # Post a message
+        cy.postMessage('Hello');
+
+        cy.getLastPostId().then((postId) => {
+            // # On a message in center channel, click then pin the post to the channel
+            cy.getPostMenu(postId, 'Pin to Channel').click();
+
+            // # Search for "Hello"
+            cy.get('#searchBox').should('be.visible').type('Hello').type('{enter}');
+
+            // * Post appears in RHS search results, displays Pinned badge
+            cy.get(`#searchResult_${postId}`).findByText('Pinned').should('exist');
+
+            // # On a message in center channel, Click [...] > Un-pin from channel
+            cy.getPostMenu(postId, 'Unpin from Channel').click();
+
+            // * Post still appears in RHS search results, but Pinned badge is removed
+            cy.get(`#searchResult_${postId}`).findByText('Pinned').should('not.exist');
+        });
+    });
+
+    it('MM-T2170 Un-pinning a post in *permalink* view also removes badge from flagged posts RHS', () => {
+        // # Post a message
+        cy.postMessage('Permalink post.');
+
+        cy.getLastPostId().then((postId) => {
+            // # On a message in center channel, click then pin the post to the channel
+            cy.getPostMenu(postId, 'Pin to Channel').click();
+
+            // # Click flag icon
+            cy.clickPostSaveIcon(postId);
+
+            // # In RHS, click Jump to view permalink view
+            cy.get('#channelHeaderFlagButton').should('exist').click();
+            cy.get(`#searchResult_${postId}`).should('exist');
+            cy.get('a.search-item__jump').first().click();
+
+            // * Message is displayed in center channel and highlighted (permalink view)
+            cy.get(`#post_${postId}`).
+                should('have.css', 'animation-duration', '1s').
+                and('have.css', 'animation-delay', '5s').
+                and('have.class', 'post--highlight');
+
+            // # In permalink view, click [...] > Un-pin from channel
+            cy.getPostMenu(postId, 'Unpin from Channel').click();
+
+            // * Pinned badge is removed in both center and RHS
+            cy.get(`#post_${postId}`).findByText('Pinned').should('not.exist');
+            cy.get(`#searchResult_${postId}`).findByText('Pinned').should('not.exist');
+        });
+    });
+
+    it('MM-T2171 Un-pinning and pinning a post in center also removes and adds badge in *flagged posts* RHS (Se portion)', () => {
+        // # Post a message
+        cy.postMessage('This is a post that is going to be pinned then removed, then pinned again.');
+
+        cy.getLastPostId().then((postId) => {
+            // # On a message in center channel, click then pin the post to the channel
+            cy.getPostMenu(postId, 'Pin to Channel').click();
```

Comment:
nit: as this selector + action is being used several times in the same fashion as `cy.clickPostSaveIcon()`, maybe it would be worth to create a `cy.clickPostPinIcon()` function as well

Reply:
This would be awesome to add!

Reply:
Done 👍